### DISCUSSION TITLE
(Dedicated) Raspberry Pi 3B enough to run this?

### DISCUSSION THREAD

**Discussion by SpuriousGer** [itsToggle/plex_debrid#389] [Q&A]
I know nothing about the requirements for such software, but I own a bunch of Raspberry Pi 3B that I could use. Would they be sufficient? What about internet connection - wifi or lan?

**Comment by itsToggle** [comment]
As long as your plex clients can play the downloaded content without transcoding, the Pi should be powerful enough. You can check out official plex sources to check which types of media your client devices can directly play and filter out any media formats that arent compatible using plex_debrids version rules.

You should definitly be using whatever connection has more bandwidth :)

**Comment by SpuriousGer** [comment]
Thanks for the rapid response. To my understanding, a Raspberry Pi 3 is usually not powerful enough to play/serve Blueray  content. But given that this is streamed, is the Pi actually doing anything besides providing connectivity?
The setup is basically like this:
- **Internet connection** is fiber (1Gbit up and down)
- **Rapsberry Pi 3B** runs Plex Debrid
- **Raspberry Pi 3B** is connected to the internet with a LAN cable, as such the connection is limited to 100Mbit due to the Pi's LAN port
- A _state of the art_ **laptop** is used to view the content.
- The **laptop** is connected to the internet either through WiFi or LAN

I would want to view the stuff in the highest resolution. I am almost certain that if the Pi was to run the setup with attached local storage, it would not work. But with this streaming like setup, I am not sure if the same impedients prevail.

**Reply by itsToggle** [reply]
Hm, so the way this setup works is that your plex client (your laptop) will request to stream a file. Your plex server will then tell rclone that it needs the first chunk of the file (125MB or something) and rclone will download this chunk from realdebrid. Going back up the chain, your Plex server will then send this chunk to your Plex client and you will be able to watch the first chunk of the movie. In the background, the Plex Server will keep requesting chunks of the file to serve your client and rclone will download these chunks as they are requested. These chunks are only saved into your PIs RAM.

This is very much like having the files locally, so if you are struggeling with local Bluray content you will most likely struggle with this setup. But Id say try it out to be sure. There are some people on discord who are running this on their PI, though Im not sure what quality and resolution they are streaming.

Have you considered a VPS like the free to use oracle cloud free-tier? There are some wonderful guides for this on the plex_debrid discord server.

**Reply by NabeelUppel** [reply]
I'm just curious have you given this a shot? I was thinking of getting myself a rp as well for this and want to get your take on it before hand

**Comment by SpuriousGer** [comment]
I try to stay as local as possible - especially trying to avoid large American cloud providers for this sort of setup.

1. I will 100% try if it works. Maybe the limitations of a Pi 3 came from something else, I haven't tested it personally, it was just something I read a while back. If the transcoding happens on the client, it shouldn't be a problem, I think.
2. It's patience vs. immediate satisfaction, I have a Pi 3 sitting around here, so I could test immediately. Whereas a beefer device, I would need to see.

Somewhat related question: have you seen anyone integrate this in Home Assistant?

**Reply by itsToggle** [reply]
Im sure there are a bunch of Plex Home Assistant integrations out there! :) Plex is quite a popular platform after all.
Plex_debrid just ties some services together (like downloading anything in your plex watchlist), so if there is a voice command available to put something into your plex watchlist, you can download new content with a voice command ;)

**Comment by SpuriousGer** [comment]
Excellent, a plex Home Assistant integration exists, so this should then work somehow as well. If I can run it on the same machine, it's actually a Pi 4 4GB and I'd be fine. Thanks for your promptness in responding, much appreciated.

**Reply by ankushg** [reply]
did you ever test this out on home-assistant or a pi 3b?
